packages/@aws-cdk/aws-config/lib/rule.ts

```diff
@@ -352,11 +353,20 @@ export class CustomRule extends RuleNew {
         messageType: 'ScheduledNotification',
       });
     }
-
-    props.lambdaFunction.addPermission('Permission', {
-      principal: new iam.ServicePrincipal('config.amazonaws.com'),
-      sourceAccount: this.env.account,
-    });
+    const hash = createHash('sha256')
+      .update(JSON.stringify({
+        fnName: props.lambdaFunction.functionName.toString,
+        accountId: this.env.account,
```

Comment:
Both account and region need to be resolved. This should fix the issue you are having with the environment agnostic integ tests.

```ts
accountId: Stack.of(this).resolve(this.env.account),
```